sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java

```diff
@@ -283,8 +283,13 @@ private void sendRequestsForBulk(CompletableFuture<BitSet> bulkResultFuture, Map
                 @Override
                 public void onResponse(ShardResponse shardResponse) {
                     nodeJobsCounter.decrement(shardLocation.nodeId);
-                    processShardResponse(shardResponse);
-                    countdown();
+                    Exception failure = shardResponse.failure();
+                    if (failure != null && failure instanceof InterruptedException) {
+                        bulkResultFuture.completeExceptionally(failure);
```

Comment:
It's not the same actually. We should completeExceptionally the resultFuture and the consumeIterator should check if the resultFuture is completed or not before continuing to consume